docs/DeveloperGuide.adoc

```diff
@@ -153,93 +155,159 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+==== Design Considerations
+
+===== Aspect: How undo & redo executes
+
+* **Alternative 1 (current choice):** Saves a copy of the  entire module planner in the `historyStateList`. It does this by performing a clone operation through the hierachy of classes of the ModulePlanner, from StudyPlan, Semester, Module down to Tag.
+** Pros: Easier to implement.
+** Cons: May have performance issues in terms of memory usage. Need to take note of the implications of cloning all the objects.
+* **Alternative 2:** Implement redo/undo for every single command.
+** Pros: Will use less memory (e.g. for `removemod`, just save the person being deleted).
+** Cons: Tedious because there are a lot of commands and we must ensure that the implementation of each individual command are correct.
+// end::undoredo[]
+
+// tag::dataencryption[]
+=== [Work in Progress] Version Tracking
+==== Current Implementation
+
+The version tracking mechanism (or _commit_ mechanism) is facilitated by `VersionTrackingManager`.
+It is stored as an instance attribute of a `ModulePlanner` object.
+Additionally, it implements the following operations:
+
+* `VersionTrackingManager#getStudyPlanCommitManagerByStudyPlan(StudyPlan studyPlan)` -- Returns the `StudyPlanCommitManager` for the specified study plan.
+* `VersionTrackingManager#commitStudyPlan(StudyPlan studyPlan, String commitMessage)` -- Saves the current state of the
+study plan in its history stored in its corresponding `StudyPlanCommitManager`, together with a commit message.
+* `VersionTrackingManager#deleteStudyPlanCommitManagerByIndex(int index)` -- Deletes a `StudyPlanCommitManager`
+specified by the index of its corresponding study plan.
+
+These operations are exposed in the `Model` interface as `Model#commitActiveStudyPlan(String commitMessage)`,
+ `Model#deleteStudyPlanCommitManagerByIndex(int index)` etc.
+
+Given below is an example usage scenario and how the commit mechanism behaves at each step.
+
+
+Step 1. The user launches the application for the first time. The `VersionTrackingManager` will be initialized with the
+initial module planner state, and its `StudyPlanCommitManagerList` will only contain a manager for the default study plan.
+
+image::VersionTrackingClassDiagram.png[]
+
+Step 2. After changing the state of the current active study plan (e.g. by calling `addmodule y1s1 cs1101s`),
+the user executes `commit first draft` command to save the current state of this study plan with the commit message
+`first draft`. The `commit` command calls `Model#commitActiveStudyPlan(String commitMessage)`,
+causing the modified state of the current active study plan after the `commit first draft command executes to be cloned and saved
+in the `StudyPlanCommitManager` corresponding to this study plan, inside the `VersionTrackingManager` under
+`ModulePlanner`.
+
+image::CommitStudyPlan.png[]
```

Comment:
The diagram and the font size in the diagram are rather too big. Would it be better if you can shrink it to fit the text?

![image](https://user-images.githubusercontent.com/46176136/67541251-dc5fff80-f71a-11e9-8052-152672ad13fd.png)


Also, you may want to include labels eg. "Figure X.. "